### ISSUE TITLE
Confusing Error Message on Bad Backup Lock Requests

### ISSUE THREAD

**Issue by jeremyk-91** [palantir/atlasdb#1709]
```
$ curl -XPOST localhost:3828/persistent-lock/acquire-backup-lock
{"code":500,"message":"There was an error processing your request. It has been logged (ID beae188edf548efb)."}
```

Ideally this should return a 400 bad request (rather than 500 internal server error) and also the message (`"Please provide a reason for acquiring the lock."`) should go to the user, I think.

Also 

```
$ curl -XPOST localhost:3828/persistent-lock/acquire-backup-lock?reason=foo
"427eb02a-f017-40cd-8d08-0a163315029a"
$ curl -XPOST localhost:3828/persistent-lock/acquire-backup-lock?reason=foo
{"exceptionClass":"com.palantir.atlasdb.keyvalue.api.CheckAndSetException","message":"Error 462f14b0-7997-4ebf-ad00-32a98ddf4b63: Check and set failed.","code":409}
```

409 tells you it's a conflict, but the error message is not great. I think from a usability POV the first one is more significant though

**Comment by tpetracca** [comment]
CC @ashrayjain

**Comment by ashrayjain** [comment]
The first issue should be fixable by changing the preconditions null check to an argument check instead. (or using `WebPreconditions` from http-remoting that does this automatically) The message part will be a bit harder to change since I believe that is controlled by http-remoting if people set up their servers with `HttpRemotingJerseyFeature.DEFAULT`. The second thing should be straightforward.

**Comment by gsheasby** [comment]
Fixed by #1909.
